File: js/849.js
/**
 * @param {number[]} seats
 * @return {number}
 */
var maxDistToClosest = function (seats) {

  const s = []
  for (let i = 0; i < seats.length; i++) {
    if (seats[i]) {
      s.push(i)
    }
  }

  s.unshift(-s[0])
  s.push(2 * (seats.length - 1) - s[s.length - 1])

  let max = 0
  for (let i = 1; i < s.length; i++) {
    const dist = parseInt((s[i] - s[i - 1]) / 2)
    if (max < dist) {
      max = dist
    }
  }

  return max
};

var assert = require('assert');
assert.equal(maxDistToClosest([1, 0, 0, 0, 1, 0, 1]), 2)
assert.equal(maxDistToClosest([1, 0, 0, 0]), 3)

File: py/849.py
class Solution:
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        s = []
        for i in range(0, len(seats)):
            if seats[i] == 1:
                s.append(i)
        s.insert(0, -s[0])
        s.append(2 * (len(seats) - 1) - s[len(s) - 1])

        # print(s)
        m = 0
        for i in range(1, len(s)):
            dist = int((s[i] - s[i-1]) / 2)
            if m < dist:
                m = dist
        return m


s = Solution()
assert(s.maxDistToClosest([1, 0, 0, 0, 1, 0, 1]) == 2)
assert(s.maxDistToClosest([1, 0, 0, 0]) == 3)
